pkg/url/url_test.go

```diff
@@ -2584,3 +2776,418 @@ func TestConvertEnvinfoURL(t *testing.T) {
+										Secure:     false,
+									},
+								},
+							},
+						},
+						{
+							Container: &versionsCommon.Container{
+								Name: "testcontainer2",
+								Endpoints: []versionsCommon.Endpoint{
+									{
+										Name:       urlName2,
+										TargetPort: 9090,
+										Secure:     true,
+										Path:       "/testpath",
+										Exposure:   versionsCommon.Internal,
+										Protocol:   versionsCommon.HTTPS,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantComponents: []versionsCommon.DevfileComponent{
+				{
+					Container: &versionsCommon.Container{
+						Image:     "quay.io/nodejs-12",
+						Name:      "testcontainer1",
+						Endpoints: []versionsCommon.Endpoint{},
+					},
+				},
+				{
+					Container: &versionsCommon.Container{
+						Name: "testcontainer2",
+						Endpoints: []versionsCommon.Endpoint{
+							{
+								Name:       urlName2,
+								TargetPort: 9090,
+								Secure:     true,
+								Path:       "/testpath",
+								Exposure:   versionsCommon.Internal,
+								Protocol:   versionsCommon.HTTPS,
+							},
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:    "Case 4: delete an invalid endpoint",
+			urlName: "invalidurl",
+			devObj: parser.DevfileObj{
+				Ctx: devfileCtx.FakeContext(fs, parser.OutputDevfileYamlPath),
+				Data: &testingutil.TestDevfileData{
+					Components: []versionsCommon.DevfileComponent{
+						{
+							Container: &versionsCommon.Container{
+								Image: "quay.io/nodejs-12",
+								Name:  "testcontainer1",
+								Endpoints: []versionsCommon.Endpoint{
+									{
+										Name:       urlName,
+										TargetPort: 8080,
+										Secure:     false,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantComponents: []versionsCommon.DevfileComponent{
+				{
+					Container: &versionsCommon.Container{
+						Image: "quay.io/nodejs-12",
+						Name:  "testcontainer1",
+						Endpoints: []versionsCommon.Endpoint{
+							{
+								Name:       urlName,
+								TargetPort: 8080,
+								Secure:     false,
+							},
+						},
+					},
+				},
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := RemoveEndpointInDevfile(tt.devObj, tt.urlName)
+			if !tt.wantErr && err != nil {
+				t.Errorf("Unexpected err from UpdateEndpointsInDevfile: %v", err)
+			} else if err == nil && tt.wantErr {
+				t.Error("error was expected, but no error was returned")
+			}
+			retval, err := fs.ReadFile(parser.OutputDevfileYamlPath)
+			if err != nil {
+				t.Errorf("Unexpected err from ReadFile: %v", err)
+			}
+			jsonval, err := devfileCtx.YAMLToJSON(retval)
+			if err != nil {
+				t.Errorf("Unexpected err from YAMLToJSON: %v", err)
+			}
+			var newData testingutil.TestDevfileData
+
+			err = json.Unmarshal(jsonval, &newData)
+			if err != nil {
+				t.Errorf("Unexpected err from json.Unmarshal: %v", err)
+			}
```

Comment:
these seem to be the same steps as above. Let's put them in a function.

Reply:
This part was for the old implementation which uses the containerEndpointMap. The latest implementation does not need this part. Deleted.